exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/DrillLimitRelBase.java

```diff
@@ -36,11 +36,17 @@
 public abstract class DrillLimitRelBase extends SingleRel implements DrillRelNode {
   protected RexNode offset;
   protected RexNode fetch;
+  private boolean pushDown;  // whether limit has been push past its child.
```

Comment:
Actually sort and aggregate have some of this behavior too. You can do a partial sort or partial aggregate in the children but then you need to combine.